static/templates/actors/partials/spell-collection.hbs

```diff
@@ -57,16 +78,21 @@
                     </div>
 
                     <div class="cast-spell"></div>
-                    {{#if @root.options.editable}}
+                    {{#if @root.editable}}
                         <div class="item-controls" data-tooltip-class="pf2e">
                             {{#unless entry.isPrepared}}
-                                <a class="item-create" data-tooltip="PF2E.CreateSpellTitle" data-type="spell"
-                                    data-level="{{section.level}}" data-location="{{entry.id}}"><i class="fa-solid fa-fw fa-plus"></i></a>
+                                <a
+                                    class="item-create"
+                                    data-tooltip="PF2E.CreateSpellTitle"
+                                    data-type="spell"
+                                    data-group-id="{{group.id}}"
+                                    data-location="{{entry.id}}"
+                                ><i class="fa-solid fa-fw fa-plus"></i></a>
                                 <a
                                     data-action="browse-spells"
                                     data-tooltip="PF2E.OpenSpellBrowserTitle"
-                                    {{#if (eq section.level 0)}}data-category="cantrip"{{/if}}
-                                    data-rank="{{#if (or (eq entry.category "ritual") (eq rank 0))}}10{{else}}{{rank}}{{/if}}"
+                                    {{#if (eq group.id "cantrips")}}data-category="cantrip"{{/if}}
+                                    data-rank="{{#if (or (eq entry.category "ritual") (eq group.id "cantrips"))}}{{entry.maxRank}}{{else}}{{group.id}}{{/if}}"
```

Comment:
It might be cleaner to have rank or castRank as part of the spell sheet data itself to clean this up a tad. The less if statements we have here the better.